src/docgen/transpile/transpile.ts

```diff
@@ -667,11 +670,16 @@ export abstract class TranspileBase implements Transpile {
   }
 
   protected getParentModule(moduleLike: reflect.ModuleLike): reflect.Assembly {
-    if (moduleLike.types.length === 0) {
+    const cached = this.parentModulesCache.get(moduleLike.fqn);
+    if (cached) return cached;
+
+    const types = moduleLike.types;
```

Comment:
This was the culprit. Apparently doing [`Object.values`](https://github.com/aws/jsii/blob/main/packages/jsii-reflect/lib/module-like.ts#L37) on a map with 8000 entries takes roughly 2.3ms, and this used to happen twice for each `typeReference` transpilation. 

Specifically for non typescript languages this is exacerbated because of parameter expansion.